### PR TITLE
Add Svelte layer

### PR DESCRIPTION
**Pull request by msuess**
This PR adds support for Svelte based on the excellent Vue layer by Thanh Vuong.

### CODE REVIEW

**Comment by msuess** [comment]
Thanks for the review! I'm assuming you mean the double newlines after the first comment? I can remove them if this is blocking the merge, but I just copied most of the code over from the vue layer, where they are also present.

File: layers/+frameworks/svelte/config.el [hunk 1]
```diff
@@ -0,0 +1,28 @@
+;;; config.el --- vue layer config file for Spacemacs. -*- lexical-binding: t -*-
```
Comment (smile13241324) [inline layers/+frameworks/svelte/config.el:1]: This should be svelte layer not vue
```suggestion
;;; config.el --- svelte layer config file for Spacemacs. -*- lexical-binding: t -*-
```

File: layers/+frameworks/svelte/config.el [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+;;
+;; Copyright (c) 2012-2021 Sylvain Benner & Contributors
+;;
+;; Author: Thanh Vuong <thanhvg@gmail.com>
```
Comment (smile13241324) [inline layers/+frameworks/svelte/config.el:5]: You should be the author :)

File: layers/+frameworks/svelte/funcs.el [hunk 3]
```diff
@@ -0,0 +1,176 @@
+;;; funcs.el --- svelte layer funcs file for Spacemacs. -*- lexical-binding: t -*-
+;;
+;; Copyright (c) 2012-2021 Sylvain Benner & Contributors
+;;
+;; Author: Thanh Vuong <thanhvg@gmail.com>
```
Comment (smile13241324) [inline layers/+frameworks/svelte/funcs.el:5]: You should be the author ;)

File: layers/+frameworks/svelte/funcs.el [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+                                   ("js" "^\\s-*\\(watch\\):\\s-?{" 1)
+                                   ("js" "^\\s-*\\(methods\\):\\s-?{" 1)
+                                   ("js" "^\\s-*\\(props\\):\\s-?{" 1)
+                                   ("css" "^<css>$" 0))
+        imenu-create-index-function #'imenu-default-create-index-function))
+
+(defun spacemacs//svelte-setup-dumb-company ()
+  (spacemacs|add-company-backends :backends (company-web-html company-css company-files company-dabbrev)
+                                  :modes svelte-mode
+                                  :variables company-minimum-prefix-length 2)
```
Comment (smile13241324) [inline layers/+frameworks/svelte/funcs.el:71]: This should rather be set in the users dotfile

File: layers/+frameworks/svelte/layers.el [hunk 5]
```diff
@@ -0,0 +1,24 @@
+;;; layers.el --- react Layer layers File for Spacemacs
```
Comment (smile13241324) [inline layers/+frameworks/svelte/layers.el:1]: Should be svelte layer

File: layers/+frameworks/svelte/layers.el [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+;;
+;; Copyright (c) 2012-2021 Sylvain Benner & Contributors
+;;
+;; Author: Thanh Vuong <thanhvg@gmail.com>
```
Comment (smile13241324) [inline layers/+frameworks/svelte/layers.el:5]: You should be the author

File: layers/+frameworks/svelte/layers.el [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+(configuration-layer/declare-layers '(node html prettier javascript))
```
Comment (smile13241324) [inline layers/+frameworks/svelte/layers.el:24]: You should declare a dependency on lsp layer when the backend lsp is used

**Review by smile13241324 — CHANGES_REQUESTED** [review]
I have reviewed your commit and found some points you need to double check before this can be merged.

**Comment by smile13241324** [comment]
Thank you for contributing to spacemacs 💜, I have cherry picked your commit into develop as I was doing some refactoring anyway I have fixed the open points on the go.

**Comment by msuess** [comment]
Thank you!

**Pull request closed without merging**